### ISSUE TITLE
networkd emits PropertiesChanged dbus signal despite SLAAC addresses remaining the same

### ISSUE THREAD

**Issue by Earnestly** [systemd/systemd#25456]
### systemd version the issue has been seen with

systemd 252 (252.1-1-arch)

### Used distribution

Arch Linux

### Linux kernel version used

6.0.9-arch1-1

### CPU architectures issue was seen on

x86_64

### Component

systemd-networkd

### Unexpected behaviour you saw

Since 252.1 I have started to notice my logs filling up with messages from `systemd-timesyncd` about the network changing at an interval of about 120s to 150s, e.g.:

```
Nov 20 19:49:24 teapot systemd-timesyncd[33641]: Network configuration changed, trying to establish connection.
Nov 20 19:49:24 teapot systemd-timesyncd[33641]: Contacted time server 185.137.172.81:123 (2.arch.pool.ntp.org).
```

After enabling debug logging for `systemd-networkd` this event appears to be caused by a Router Advertisement.  damjan(irc) suggested that `systemd-networkd` probably ought not emit a dbus signal if the resulting SLAAC address remains unchanged.

Below is an example of the debug output I see:

<details><summary>Sample <code>systemd-networkd</code> debug log output</summary>

```
Nov 20 19:49:24 teapot systemd-networkd[33776]: enp0s25: NDISC: Received Router Advertisement: flags none preference medium lifetime 0 sec
Nov 20 19:49:24 teapot systemd-networkd[33776]: enp0s25: NDISC: Invoking callback for 'router' event.
Nov 20 19:49:24 teapot systemd-networkd[33776]: enp0s25: Requesting NDisc route (configuring,configured): dst: fdd3:79ec:333::/64, src: n/a, gw: n/a, prefsrc: n/a, scope: global, 
table: main(254), proto: ra, type: unicast, nexthop: 0, priority: 1024, flags: n/a
Nov 20 19:49:24 teapot systemd-networkd[33776]: enp0s25: Requesting NDisc address (configured): fdd3:79ec:333:0:f2de:f1ff:fe73:b2eb/64 (valid for 1h 59min 59s, preferred for 29min
 59s), flags: manage-temporary-address,no-prefixroute, scope: global
Nov 20 19:49:24 teapot systemd-networkd[33776]: enp0s25: Requesting NDisc route (configuring,configured): dst: fdd3:79ec:333::/48, src: n/a, gw: fe80::20f:94ff:fed5:c311, prefsrc:
 n/a, scope: global, table: main(254), proto: ra, type: unicast, nexthop: 0, priority: 1024, flags: n/a
Nov 20 19:49:24 teapot systemd-networkd[33776]: enp0s25: Setting SLAAC addresses and router.
Nov 20 19:49:24 teapot systemd-networkd[33776]: enp0s25: State changed: configured -> configuring
Nov 20 19:49:24 teapot systemd-networkd[33776]: Sent message type=signal sender=n/a destination=n/a path=/org/freedesktop/network1/link/_32 interface=org.freedesktop.DBus.Properties member=PropertiesChanged cookie=29 reply_cookie=0 signature=sa{sv}as error-name=n/a error-message=n/a
Nov 20 19:49:24 teapot systemd-networkd[33776]: enp0s25: link_check_ready(): IPv4LL:no DHCPv4:yes DHCPv6:no DHCP-PD:no NDisc:no
Nov 20 19:49:24 teapot systemd-networkd[33776]: enp0s25: State changed: configuring -> configured
Nov 20 19:49:24 teapot systemd-networkd[33776]: Sent message type=signal sender=n/a destination=n/a path=/org/freedesktop/network1/link/_32 interface=org.freedesktop.DBus.Properties member=PropertiesChanged cookie=30 reply_cookie=0 signature=sa{sv}as error-name=n/a error-message=n/a
Nov 20 19:49:24 teapot systemd-timesyncd[33641]: Network configuration changed, trying to establish connection.
Nov 20 19:49:24 teapot systemd-networkd[33776]: enp0s25: Configuring NDisc route (requesting,configuring,configured): dst: fdd3:79ec:333::/64, src: n/a, gw: n/a, prefsrc: n/a, scope: global, table: main(254), proto: ra, type: unicast, nexthop: 0, priority: 1024, flags: n/a
Nov 20 19:49:24 teapot systemd-networkd[33776]: enp0s25: Configuring NDisc address (requesting,configured): fdd3:79ec:333:0:f2de:f1ff:fe73:b2eb/64 (valid for 1h 59min 59s, preferred for 29min 59s), flags: manage-temporary-address,no-prefixroute, scope: global
Nov 20 19:49:24 teapot systemd-networkd[33776]: enp0s25: Configuring NDisc route (requesting,configuring,configured): dst: fdd3:79ec:333::/48, src: n/a, gw: fe80::20f:94ff:fed5:c311, prefsrc: n/a, scope: global, table: main(254), proto: ra, type: unicast, nexthop: 0, priority: 1024, flags: n/a
Nov 20 19:49:24 teapot systemd-networkd[33776]: enp0s25: ndisc_check_ready(): SLAAC addresses and routes are not set.
Nov 20 19:49:24 teapot systemd-networkd[33776]: enp0s25: Received updated NDisc address (configured): fdd3:79ec:333:0:f2de:f1ff:fe73:b2eb/64 (valid for 1h 59min 59s, preferred for 29min 59s), flags: manage-temporary-address,no-prefixroute, scope: global
Nov 20 19:49:24 teapot systemd-networkd[33776]: enp0s25: ndisc_check_ready(): SLAAC addresses and routes are not set.
Nov 20 19:49:24 teapot systemd-networkd[33776]: enp0s25: SLAAC addresses and routes set.
Nov 20 19:49:24 teapot systemd-timesyncd[33641]: Contacted time server 185.137.172.81:123 (2.arch.pool.ntp.org).
Nov 20 19:51:55 teapot systemd-networkd[33776]: enp0s25: NDISC: Received Router Advertisement: flags none preference medium lifetime 0 sec
Nov 20 19:51:55 teapot systemd-networkd[33776]: enp0s25: NDISC: Invoking callback for 'router' event.
Nov 20 19:51:55 teapot systemd-networkd[33776]: enp0s25: Requesting NDisc route (configuring,configured): dst: fdd3:79ec:333::/64, src: n/a, gw: n/a, prefsrc: n/a, scope: global, table: main(254), proto: ra, type: unicast, nexthop: 0, priority: 1024, flags: n/a
Nov 20 19:51:55 teapot systemd-networkd[33776]: enp0s25: Requesting NDisc address (configured): fdd3:79ec:333:0:f2de:f1ff:fe73:b2eb/64 (valid for 1h 59min 59s, preferred for 29min 59s), flags: manage-temporary-address,no-prefixroute, scope: global
Nov 20 19:51:55 teapot systemd-networkd[33776]: enp0s25: Requesting NDisc route (configuring,configured): dst: fdd3:79ec:333::/48, src: n/a, gw: fe80::20f:94ff:fed5:c311, prefsrc: n/a, scope: global, table: main(254), proto: ra, type: unicast, nexthop: 0, priority: 1024, flags: n/a
Nov 20 19:51:55 teapot systemd-networkd[33776]: enp0s25: Setting SLAAC addresses and router.
Nov 20 19:51:55 teapot systemd-networkd[33776]: enp0s25: State changed: configured -> configuring
Nov 20 19:51:55 teapot systemd-networkd[33776]: Sent message type=signal sender=n/a destination=n/a path=/org/freedesktop/network1/link/_32 interface=org.freedesktop.DBus.Properties member=PropertiesChanged cookie=31 reply_cookie=0 signature=sa{sv}as error-name=n/a error-message=n/a
Nov 20 19:51:55 teapot systemd-networkd[33776]: enp0s25: link_check_ready(): IPv4LL:no DHCPv4:yes DHCPv6:no DHCP-PD:no NDisc:no
Nov 20 19:51:55 teapot systemd-networkd[33776]: enp0s25: State changed: configuring -> configured
Nov 20 19:51:55 teapot systemd-networkd[33776]: Sent message type=signal sender=n/a destination=n/a path=/org/freedesktop/network1/link/_32 interface=org.freedesktop.DBus.Properties member=PropertiesChanged cookie=32 reply_cookie=0 signature=sa{sv}as error-name=n/a error-message=n/a
Nov 20 19:51:55 teapot systemd-timesyncd[33641]: Network configuration changed, trying to establish connection.
Nov 20 19:51:55 teapot systemd-networkd[33776]: enp0s25: Configuring NDisc route (requesting,configuring,configured): dst: fdd3:79ec:333::/64, src: n/a, gw: n/a, prefsrc: n/a, scope: global, table: main(254), proto: ra, type: unicast, nexthop: 0, priority: 1024, flags: n/a
Nov 20 19:51:55 teapot systemd-networkd[33776]: enp0s25: Configuring NDisc address (requesting,configured): fdd3:79ec:333:0:f2de:f1ff:fe73:b2eb/64 (valid for 1h 59min 59s, preferred for 29min 59s), flags: manage-temporary-address,no-prefixroute, scope: global
Nov 20 19:51:55 teapot systemd-networkd[33776]: enp0s25: Configuring NDisc route (requesting,configuring,configured): dst: fdd3:79ec:333::/48, src: n/a, gw: fe80::20f:94ff:fed5:c311, prefsrc: n/a, scope: global, table: main(254), proto: ra, type: unicast, nexthop: 0, priority: 1024, flags: n/a
Nov 20 19:51:55 teapot systemd-networkd[33776]: enp0s25: ndisc_check_ready(): SLAAC addresses and routes are not set.
Nov 20 19:51:55 teapot systemd-networkd[33776]: enp0s25: Received updated NDisc address (configured): fdd3:79ec:333:0:f2de:f1ff:fe73:b2eb/64 (valid for 1h 59min 59s, preferred for 29min 59s), flags: manage-temporary-address,no-prefixroute, scope: global
Nov 20 19:51:55 teapot systemd-networkd[33776]: enp0s25: ndisc_check_ready(): SLAAC addresses and routes are not set.
Nov 20 19:51:55 teapot systemd-networkd[33776]: enp0s25: SLAAC addresses and routes set.
Nov 20 19:51:55 teapot systemd-timesyncd[33641]: Contacted time server 185.137.172.81:123 (2.arch.pool.ntp.org).
```

</details>

**Comment by yuwata** [comment]
Could you provide the relevant .network file for the interface?

**Comment by Earnestly** [comment]
Nothing complicated:

```
$ tail -n +0 /etc/systemd/network/01-wire*
==> /etc/systemd/network/01-wired.network <==
[Match]
Name=en*

[Network]
DHCP=yes
Gateway=192.168.1.254

==> /etc/systemd/network/01-wireless.network <==
[Match]
Name=wl*

[Network]
DHCP=yes
Gateway=192.168.1.254
```

**Comment by topcat001** [comment]
I have just a wired network with DHCP on Arch Linux (fully updated with `systemd 252 (252.1-1-arch)`)  on a desktop, with an identical `01-wired.network` as @Earnestly above and I see the same log spam. My router is also doing advertisements.

**Comment by yuwata** [comment]
Interesting. I cannot reproduce it...

**Comment by yuwata** [comment]
I found the root cause. Fix is waiting in #25569.
